### PR TITLE
feat: add etcd 3.4.18

### PR DESCRIPTION
**Pull request by ronething**
debug actions: https://github.com/api7/etcd-rpm/actions/runs/4750254811/jobs/8438265922

branch tree: https://github.com/api7/etcd-rpm/tree/dev/etcd/3.4.18

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+      - name: run test case
+        run: |
+          export VERSION=${{ steps.etcd_version.outputs.version }}
+          ./scripts/etcd_test.sh ${VERSION}
+
+      - name: Publish Artifact
+        uses: actions/upload-artifact@v2.2.3
+        with:
+          name: "etcd-rpm"
+          path: "./etcd-${{ steps.etcd_version.outputs.version }}.x86_64.rpm"
```
Comment (nic-chen) [inline .github/workflows/build.yaml]: maybe we could upload it to the release

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+concurrency:
+  group: auto-build-etcd-rpm-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  auto_build_etcd_rpm:
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'ci-build') ||
+      contains(github.event.pull_request.labels.*.name, 'ci-all') ||
+      github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/') }}
```
Comment (starsz) [inline .github/workflows/build.yaml]: I think we can remove this. Since there is only on workflow in the project.

File: etcd.conf.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# This is the configuration file for the etcd server.
+
+# Human-readable name for this member.
+name: 'poc-cluster'
```
Comment (starsz) [inline etcd.conf.yaml]: Should remove "poc" .

File: .github/workflows/build.yaml (hunk 1 above)
Comment (ronething) [reply .github/workflows/build.yaml]: i remove this step and create a new step to upload rpm to release.

image

File: etcd.conf.yaml (hunk 3 above)
Comment (ronething) [reply etcd.conf.yaml]: change to `etcd-cluster`

File: .github/workflows/build.yaml (hunk 2 above)
Comment (ronething) [reply .github/workflows/build.yaml]: removed.

File: README.md [hunk 4]
```diff
@@ -1 +1,9 @@
-# etcd-rpm
\ No newline at end of file
+# etcd-rpm
+
+## Project Description
+
+This repository provides a simple solution for packaging etcd as an RPM package using nfpm. With the help of GitHub Actions, the RPM package is automatically built and published as a release artifact.
+
```
Comment (nic-chen) [inline README.md:6]: better to add description for `how to release a new version`

Comment (ronething) [reply README.md:6]: updated.

**Review by nic-chen — APPROVED** [review]

File: .github/workflows/build.yaml [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+      - name: run test case
+        run: |
+          export VERSION=${{ steps.etcd_version.outputs.version }}
+          ./scripts/etcd_test.sh ${VERSION}
+
+      - name: Create GitHub Release and Upload RPM
+        uses: softprops/action-gh-release@v1
+        # Only triggered on new tag
+        # if: startsWith(github.ref, 'refs/tags/')
```
Comment (nic-chen) [inline .github/workflows/build.yaml]:
```suggestion
        if: startsWith(github.ref, 'refs/tags/')
```

**Review by starsz — APPROVED** [review]

**Pull request merged**